packages/fakta-fordel-beregningsgrunnlag/src/components/tilkommetAktivitet/TilkommetAktivitetPanel.tsx

```diff
@@ -96,6 +111,75 @@ const TilkommetAktivitetPanel: FC<TilkommetAktivitetPanelType> = ({
     );
   };
 
+  const mapNyAndel = (andel: any, taMedAlleFelter: boolean, periodeFieldIndex: number, andelFieldIndex: number) => {
+    const skalRedusereValg = watch(
+      `${formName}.${bgIndex}.perioder.${periodeFieldIndex}.andeler.${andelFieldIndex}.skalRedusereUtbetaling`,
+    );
+    const bruttoValg = watch(
+      `${formName}.${bgIndex}.perioder.${periodeFieldIndex}.andeler.${andelFieldIndex}.bruttoInntektPrÅr`,
+    );
+    return {
+      aktivitetStatus: andel.aktivitetStatus,
+      arbeidsgiverIdent: andel.arbeidsgiverIdent,
+      arbeidsforholdId: andel.arbeidsforholdId,
+      bruttoInntektPrÅr: taMedAlleFelter ? bruttoValg : undefined,
+      skalRedusereUtbetaling: taMedAlleFelter ? skalRedusereValg : undefined,
+    };
+  };
+
+  const finnNyePerioder = useCallback(
+    (gammelFom: string, nyFom: string): Periode[] => {
```

Comment:
Er du egentlig avhengig av å ta inn gammelFom her? kunne du ikkje ha funnet perioden som overlapper med nyFom?

Reply:
Prøver berre å forstå koden bedre. Du er jo avhengig av å ta inn perioden i datovelgeren uansett da.

Reply:
Jeg trenger strengt tatt ikke gammel fom nei, jeg kan bare finne perioden som overlapper som du sier. Det virket bare mer forståelig for meg med en metode som tok inn gammelFom i tillegg som en indikasjon på hva som skal splittes.

Reply:
Men om dette er forvirrende / vanskelig å forstå bør vi kanskje revurdere